test/unit/picolibrary/output_stream/main.cc

```diff
@@ -502,6 +581,269 @@ TEST( putSignedByteBlock, worksProperly )
     EXPECT_TRUE( stream.is_nominal() );
 }
 
+/**
+ * \brief Verify picolibrary::Output_Stream::print() properly handles the presence of an
+ *        I/O error and/or fatal error.
+ */
+TEST( print, errorPresent )
+{
+    {
+        auto stream = Mock_Output_Stream{};
+
+        stream.report_random_error();
+
+        auto const io_error_present    = stream.io_error_present();
+        auto const fatal_error_present = stream.fatal_error_present();
+
+        EXPECT_CALL( stream.buffer(), put( A<char>() ) ).Times( 0 );
+
+        auto const result = stream.print( random_container<std::string>().c_str() );
+
+        EXPECT_TRUE( result.is_error() );
+        EXPECT_EQ( result.error(), Generic_Error::IO_STREAM_DEGRADED );
+
+        EXPECT_FALSE( stream.end_of_file_reached() );
+        EXPECT_EQ( stream.io_error_present(), io_error_present );
+        EXPECT_EQ( stream.fatal_error_present(), fatal_error_present );
+    }
+
+    {
+        auto stream = Mock_Output_Stream{};
+
+        stream.report_random_error();
+
+        auto const io_error_present    = stream.io_error_present();
+        auto const fatal_error_present = stream.fatal_error_present();
+
+        EXPECT_CALL( stream.buffer(), put( A<char>() ) ).Times( 0 );
+
+        auto const result = stream.print(
+            ( random_container<std::string>() + "{}" + random_container<std::string>() ).c_str(),
+            random<Foo>() );
+
+        EXPECT_TRUE( result.is_error() );
+        EXPECT_EQ( result.error(), Generic_Error::IO_STREAM_DEGRADED );
+
+        EXPECT_FALSE( stream.end_of_file_reached() );
+        EXPECT_EQ( stream.io_error_present(), io_error_present );
+        EXPECT_EQ( stream.fatal_error_present(), fatal_error_present );
+    }
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::print() properly handles a print error.
+ */
+TEST( print, printError )
+{
+    {
+        auto stream = Mock_Output_Stream{};
+
+        auto const error = random<Mock_Error>();
+
+        EXPECT_CALL( stream.buffer(), put( A<char>() ) ).WillOnce( Return( error ) );
+
+        auto const result = stream.print( random_container<std::string>().c_str() );
+
+        EXPECT_TRUE( result.is_error() );
+        EXPECT_EQ( result.error(), error );
+
+        EXPECT_FALSE( stream.end_of_file_reached() );
+        EXPECT_FALSE( stream.io_error_present() );
+        EXPECT_TRUE( stream.fatal_error_present() );
+    }
+
+    {
+        auto stream = Mock_Output_Stream{};
+
+        auto const error = random<Mock_Error>();
+
+        EXPECT_CALL( stream.buffer(), put( A<char>() ) ).WillOnce( Return( error ) );
+
+        auto const result = stream.print(
+            ( random_container<std::string>() + "{}" + random_container<std::string>() ).c_str(),
```

Comment:
Minimum string length needs to be at least 1 for the first string to ensure `put()` is called at least once.